src/Shared/FileUtilities.cs

```diff
@@ -485,6 +485,42 @@ internal static string MaybeAdjustFilePath(string value, string baseDirectory =
             return shouldAdjust ? newValue.ToString() : value;
         }
 
+#if !CLR2COMPATIBILITY
+        /// <summary>
+        /// If on Unix, convert backslashes to slashes for strings that resemble paths.
+        /// This overload takes and returns ReadOnlyMemory of characters.
+        /// </summary>
+        internal static ReadOnlyMemory<char> MaybeAdjustFilePath(ReadOnlyMemory<char> value, string baseDirectory = "")
+        {
+            if (NativeMethodsShared.IsWindows || value.IsEmpty)
+            {
+                return value;
+            }
+
+            // Don't bother with arrays or properties or network paths.
+            if (value.Length >= 2)
+            {
+                var span = value.Span;
+                if ((span[1] == '(' && (span[0] == '$' || span[0] == '@')) ||
```

Comment:
This looks a little faster than in the method above, but it's a little harder to understand what it's trying to do, since $ on its own doesn't scream property at me, etc. Not saying you should change it; just making a note.